### ISSUE TITLE
Image of docker-asciidoctor isn't updated in docker hub

### ISSUE THREAD

**Issue by juracy** [asciidoctor/docker-asciidoctor#125]
Maybe it's the same issue #115 

The last update was in Nov, 21 2019

**Comment by dduportal** [comment]
👍 It is weird, the "step" responsible for calling the DockerHub builder is not working anymore.
Thanks for letting us know, I'll let you now the fix

**Comment by dduportal** [comment]
So the webhook is correctly sent from Travis to the Dockerhub, which answers `HTTP/202` but never start the build:

```
make deploy
curl --verbose --header "Content-Type: application/json" \
                --data '{"source_type": "Branch", "source_name": "HEAD"}' \
                -X POST https://hub.docker.com/api/build/v1/source/23eb39b3-d423-44c4-a84d-e77d4b0e3858/trigger/566f7876-a61d-44de-a7d3-a9ffc6d9a000/call/
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying 34.232.150.6...
* TCP_NODELAY set
* Connected to hub.docker.com (34.232.150.6) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: CN=*.docker.com
*  start date: Jul 16 00:00:00 2019 GMT
*  expire date: Aug 16 12:00:00 2020 GMT
*  subjectAltName: host "hub.docker.com" matched cert's "*.docker.com"
*  issuer: C=US; O=Amazon; OU=Server CA 1B; CN=Amazon
*  SSL certificate verify ok.
> POST /api/build/v1/source/23eb39b3-d423-44c4-a84d-e77d4b0e3858/trigger/566f7876-a61d-44de-a7d3-a9ffc6d9a000/call/ HTTP/1.1
> Host: hub.docker.com
> User-Agent: curl/7.64.1
> Accept: */*
> Content-Type: application/json
> Content-Length: 48
> 
* upload completely sent off: 48 out of 48 bytes
< HTTP/1.1 202 ACCEPTED
< Date: Sun, 09 Feb 2020 08:08:27 GMT
< Content-Type: application/json
< Content-Length: 470
< X-RateLimit-Remaining: 58
< X-XSS-Protection: 1; mode=block
< X-Content-Type-Options: nosniff
< Vary: Accept
< X-RateLimit-Limit: 60
< X-Frame-Options: deny
< X-RateLimit-Reset: 54
< Server: nginx
< Strict-Transport-Security: max-age=31536000
< 
* Connection #0 to host hub.docker.com left intact
{"autotests": "OFF", "build_in_farm": true, "build_settings": ["/api/build/v1/setting/a2b51650-1019-49f9-98d0-0093d8e3f8f0/"], "channel": "Stable", "deploykey": "", "envvars": [], "image": "asciidoctor/docker-asciidoctor", "owner": "asciidoctor", "provider": "Github", "repo_links": true, "repository": "docker-asciidoctor", "resource_uri": "/api/build/v1/source/23eb39b3-d423-44c4-a84d-e77d4b0e3858/", "state": "Success", "uuid": "23eb39b3-d423-44c4-a84d-e77d4b0e3858"}* Closing connection 0
```

**Comment by dduportal** [comment]
There are issues on the Dockerhub side, which API acknowledged the build trigger events, but never act acocrding to.

**Comment by dduportal** [comment]
- [x] New Trigger URL created in DockerHub
- [x] Running commande `make deploy` locally with this new Trigger URL for initial validation: https://hub.docker.com/repository/registry-1.docker.io/asciidoctor/docker-asciidoctor/builds/3103d60d-746f-48c7-bf7e-945ce2cf8066
- [x] New Trigger URL encrypted in Travis-CI
- [x] Travis CI build launched manually on the `master` branch for validation: https://hub.docker.com/repository/registry-1.docker.io/asciidoctor/docker-asciidoctor/builds/ebeff548-2899-4569-a72b-36e349153081
- [x] Opened a PR (#127 ) and merged it to verify that the build is triggered after merge: https://hub.docker.com/repository/registry-1.docker.io/asciidoctor/docker-asciidoctor/builds/5ecf3c96-49c3-4756-b7c2-a2d93edc000b

**Comment by dduportal** [comment]
=> This 2nd happening of this event is raising questions about the reliability of the "Build Trigger" of DockerHub.

- Should we switch to docker pushing the built images from the CI ?
  - Pros: we have a feedback if it fails, and what we built and tested is what user gets at the exact byte
  - Cons: we have to encode a technical user for Docker Hub in the CI so `docker login / docker push` commands works, we have to set up a mechanism for the `README.md` update with the right tools
- Should we use the DockerHub "AutoBuild / Autotest" (e.g.replacing TravisCI by Dockerhub)?
  - Pros: full Dockerhub system
  - Cons: Not sure about the long term reliability (we are sure that the registry is ok, but what about their build system?) and usages (does it give a feedback in github ? Does is allow the `README.adoc` -> `README.md` conversion? Does it support upcoing multi stage build ?)

- Other ideas ? (ping @mojavelinux )

**Comment by dduportal** [comment]
Hey @juracy , it should be fixed (and will be "secured" by #128 ).

If it is ok for you, can you close issue, or comment otherwise?

Thanks a lot !

**Comment by juracy** [comment]
Nice!
